kedro/framework/cli/project.py

```diff
@@ -218,17 +206,53 @@ def build_docs(metadata: ProjectMetadata, open_docs):
 
 
 @forward_command(project_group, name="build-reqs")
+@click.option(
+    "--input-file",
+    "input_file",
+    multiple=False,
```

Comment:
`click` has a handy `click.Path` type we can use for validation. We use it for other file arguments/options as well.

```python
type=click.Path(exists=True, dir_okay=False, resolve_path=True),
```

That way we don't have to do our own error handling, but leave it up to `click`. Can't remember what gets outputted, if it's a nice error message it's probably easier to rely on that instead.

Reply:
TIL! 👍  
However, this won't catch the case where the user hasn't specified an input file and there isn't a `requirements.txt` in the source directory in the same way. In that case you'd see:
```
Error: Invalid value for '[SRC_FILES]...': Path '/Users/merel_theisen/Projects/Testing/reqs/src/requirements.txt' does not exist.
```

Our own custom error might be a bit more informative. WDYT?

Reply:
I think this case is very unlikely given that projects come with `src/requirements.txt` and there's not really any reason someone would remove it.

Even better though, could we use `default` in the `click.option` to set the default input and output file paths? That way click will check the existence of `input-path` even if it's the default path (though obviously you don't want to do that check for `output-path`) and you could remove 
```
    input_file = input_file or source_path / "requirements.txt"
    output_file = output_file or source_path / "requirements.lock"
```

Not a biggy either way though 🙂

Reply:
I'm don't think you can set the default path because we need to calculate the `source_path` first.